Xamarin.Forms.Platform.UAP/Platform.cs

```diff
@@ -561,9 +561,25 @@ internal static void SubscribeAlertsAndActionSheets()
 			MessagingCenter.Subscribe<Page, PromptArguments>(Window.Current, Page.PromptSignalName, OnPagePrompt);
 		}
 
-		static void OnPageActionSheet(object sender, ActionSheetArguments options)
+		static void OnPageActionSheet(Page sender, ActionSheetArguments options)
 		{
 			bool userDidSelect = false;
+
+			if (options.FlowDirection == FlowDirection.MatchParent)
+			{
+				if (sender.FlowDirection == FlowDirection.RightToLeft)
+					options.FlowDirection = FlowDirection.RightToLeft;
+				else if (sender.FlowDirection == FlowDirection.LeftToRight)
+					options.FlowDirection = FlowDirection.LeftToRight;
+				else
+				{
+					if (Device.FlowDirection == FlowDirection.RightToLeft)
+						options.FlowDirection = FlowDirection.RightToLeft;
+					else if (Device.FlowDirection == FlowDirection.LeftToRight)
+						options.FlowDirection = FlowDirection.LeftToRight;
+				}
+			}
```

Comment:
Instead of checking the `FlowDirection` values for the Page (and then the Device) here, I think you can just use the Page's `EffectiveFlowDirection` - it should already be set to either `RightToLeft` or `LeftToRight` (depending on its `FlowDirection` settings.